src/PureScript/Backend/Optimizer/Analysis.purs

```diff
@@ -338,6 +338,8 @@ analyze externAnalysis expr = case expr of
     analyzeDefault expr
   Accessor hd acc ->
     case syntaxOf hd of
+      Just (Accessor _ (GetCtorField qi _ _ _ _ _)) ->
+        analysis <> usedDep qi
       Just (Accessor _ _) ->
```

Comment:
This is casing the wrong thing. It should be casing on `acc` in the `where` block below.

Reply:
Can you clarify what "tests break" mean? Snapshots changing may be expected without further work.

Reply:
Here is one example:

```
✗ Snapshot.RecursionSchemes01 failed.
15a16,21
> const $$eval = v => {
>   if (v.tag === "Add") { return v._1 + v._2 | 0; }
>   if (v.tag === "Mul") { return v._1 * v._2 | 0; }
>   if (v.tag === "Lit") { return v._1; }
>   $runtime.fail();
> };
17,20c23,26
<   const go = v => {
<     if (v.tag === "Add") { return go(v._1) + go(v._2) | 0; }
<     if (v.tag === "Mul") { return go(v._1) * go(v._2) | 0; }
<     if (v.tag === "Lit") { return v._1; }
---
>   const go = v => $$eval((() => {
>     if (v.tag === "Add") { return $ExprF("Add", go(v._1), go(v._2)); }
>     if (v.tag === "Mul") { return $ExprF("Mul", go(v._1), go(v._2)); }
>     if (v.tag === "Lit") { return $ExprF("Lit", v._1); }
22c28
<   };
---
>   })());
27,30c33,39
<     if (v.tag === "Add") { return go(v._1) + go(v._2) | 0; }
<     if (v.tag === "Mul") { return go(v._1) * go(v._2) | 0; }
<     if (v.tag === "Lit") { return v._1 + 1 | 0; }
<     $runtime.fail();
---
>     const $0 = (() => {
>       if (v.tag === "Add") { return $ExprF("Add", go(v._1), go(v._2)); }
>       if (v.tag === "Mul") { return $ExprF("Mul", go(v._1), go(v._2)); }
>       if (v.tag === "Lit") { return $ExprF("Lit", v._1); }
>       $runtime.fail();
>     })();
>     return $$eval($0.tag === "Lit" ? $ExprF("Lit", $0._1 + 1 | 0) : $0);
34c43
< export {$ExprF, Add, Lit, Mul, functorExprF, test1, test2};
---
> export {$ExprF, Add, Lit, Mul, $$eval as eval, functorExprF, test1, test2};
```

Reply:
This is from the change I now pushed/